client/src/components/FieldHolder/FieldHolder.js

```diff
@@ -123,26 +143,34 @@ function fieldHolder(Field) {
   }
 
   FieldHolder.propTypes = {
-    leftTitle: React.PropTypes.any,
-    rightTitle: React.PropTypes.any,
-    title: React.PropTypes.any,
-    extraClass: React.PropTypes.string,
-    holderId: React.PropTypes.string,
-    id: React.PropTypes.string,
-    description: React.PropTypes.any,
-    hideLabels: React.PropTypes.bool,
-    message: React.PropTypes.shape({
-      extraClass: React.PropTypes.string,
-      value: React.PropTypes.any,
-      type: React.PropTypes.string,
+    leftTitle: PropTypes.any,
+    rightTitle: PropTypes.any,
+    title: PropTypes.any,
+    extraClass: PropTypes.string,
+    holderId: PropTypes.string,
+    id: PropTypes.string,
+    description: PropTypes.any,
+    hideLabels: PropTypes.bool,
+    message: PropTypes.shape({
+      extraClass: PropTypes.string,
+      value: PropTypes.any,
+      type: PropTypes.string,
     }),
+    data: PropTypes.oneOfType([
+      PropTypes.array,
```

Comment:
The trouble stems from php converting empty associative arrays into *gasp* arrays :(
So where originally it could be PropTypes.shape alone, if none of the fields inside are required, it could be returned to us as an array...